server/models/procedures/stock.sql

```diff
@@ -117,23 +123,47 @@ BEGIN
   CREATE TEMPORARY TABLE stage_stock_movement (
       SELECT
         projectId as project_id, currencyId as currency_id,
+        CONCAT(ig.name, ' - ', m.quantity, ' ', iu.text, ' of ', i.text , ' (', l.label, ')') AS item_description,
         m.uuid, m.description, m.date, m.flux_id, m.is_exit, m.document_uuid, m.quantity, m.unit_cost, m.user_id,
-        ig.cogs_account, ig.stock_account
+        ig.cogs_account, ig.stock_account, d.text AS depot_description,
+        CONCAT(p.display_name, '(', em.text, ')') AS patient_description,
+        s.name AS service_description, dm.text AS invoice_description
       FROM stock_movement m
+      JOIN depot d ON d.uuid = m.depot_uuid
       JOIN lot l ON l.uuid = m.lot_uuid
       JOIN inventory i ON i.uuid = l.inventory_uuid
+      JOIN inventory_unit iu ON iu.id = i.unit_id
       JOIN inventory_group ig
         ON ig.uuid = i.group_uuid AND (ig.stock_account IS NOT NULL AND ig.cogs_account IS NOT NULL)
+      LEFT JOIN patient p ON p.uuid = m.entity_uuid
+      LEFT JOIN entity_map em ON em.uuid = p.uuid
+      LEFT JOIN service s ON s.uuid = m.entity_uuid 
+      LEFT JOIN document_map dm ON dm.uuid = m.invoice_uuid
       WHERE m.document_uuid = documentUuid AND m.is_exit = isExit
     );
 
   -- define voucher variables
-  SELECT HUID(UUID()), date, project_id, currency_id, user_id, description, SUM(unit_cost * quantity)
-    INTO voucher_uuid, voucher_date, voucher_project_id, voucher_currency_id, voucher_user_id, voucher_description, voucher_amount
+  SELECT HUID(UUID()), date, project_id, currency_id, user_id, description, SUM(unit_cost * quantity),
+    depot_description, patient_description, service_description, invoice_description
+    INTO voucher_uuid, voucher_date, voucher_project_id, voucher_currency_id, voucher_user_id, voucher_description, voucher_amount,
+      v_depot_description, v_patient_description, v_service_description, v_invoice_description
   FROM stage_stock_movement;
 
   IF (isExit = 1) THEN
     SET voucher_type_id = STOCK_EXIT_TYPE;
+    SET v_voucher_description = voucher_description;
+
+    IF (v_depot_description IS NOT NULL AND v_service_description IS NOT NULL) THEN 
+      SET voucher_description = CONCAT('Distribution to ', v_service_description, ' from ', v_depot_description, ' : ', v_voucher_description);
+    END IF;
+
+    IF (v_depot_description IS NOT NULL AND v_patient_description IS NOT NULL) THEN 
+      SET voucher_description = CONCAT('Distribution to ', v_patient_description, ' from ', v_depot_description, ' : ', v_voucher_description);
+    END IF;
+    
+    IF (v_depot_description IS NOT NULL AND v_patient_description IS NOT NULL AND v_invoice_description IS NOT NULL) THEN 
```

Comment:
Oof.  This is super complicated.  Can't we find a way to do this in JavaScript to get regular translation support?  For example, this is how the patient invoice description is created: https://github.com/IMA-WorldHealth/bhima/blob/master/client/src/i18n/en/patient_invoice.json#L15.

Reply:
We must add a server side api end point for getting details of the description, and before to create a stock movement we must ask the API end point for having description details and after that we can create the stock movement on the client side.